LICENSE-binary

```diff
@@ -535,6 +525,24 @@ jakarta.ws.rs-api:jakarta.ws.rs-api:2.1.6
 jakarta.servlet.jsp:jakarta.servlet.jsp-api:2.3.6
 jakarta.servlet:jakarta.servlet-api:4.0.4
 
+Eclipse Public License (EPL) 2.0 with some parts being
+GNU General Public License (GPL), Version 2, With Classpath Exception,
+Apache License, 2.0 or Public Domain
+--------------------------
+
+See also https://github.com/eclipse-ee4j/jersey/blob/master/NOTICE.md
+
+org.glassfish.jersey.core:jersey-common:2.46
+org.glassfish.jersey.core:jersey-server:2.46
+org.glassfish.jersey.inject:jersey-hk2:2.46
+org.glassfish.jersey.core:jersey-client:2.46
+org.glassfish.jersey.test-framework:jersey-test-framework-core:2.46
+org.glassfish.jersey.test-framework.providers:jersey-test-framework-provider-grizzly2:2.46
+org.glassfish.jersey.containers:jersey-container-servlet:2.46
+org.glassfish.jersey.containers:jersey-container-servlet-core:2.46
+org.glassfish.jersey.media:jersey-media-json-jettison:2.46
+org.glassfish.jersey.media:jersey-media-jaxb:2.46
+
```

Comment:
@steveloughran @ayushtkn @Hexiaoqiao Using this PR, I would like to ask whether we need to backport this to branch-3.4, or should we release it in hadoop-3.5.0?

Reply:
I'm starting to wonder what it'd take to get a 3.5.0 out, with java 17 the minimum version...

how incompatible is this change downstream?

Reply:
From my perspective, I haven't noticed any significant compatibility issues so far, as Jersey 2.46 should be able to compile without issues on JDK 8, JDK 11, and JDK 17. I tried compiling the trunk code with JDK 17, and everything worked fine. Of course, if we plan to release the first version officially supporting JDK 17, there is still some additional work to be done. The biggest task right now is upgrading from JUnit 4 to JUnit 5, and my colleague and I are working on this together. He has developed an automated code conversion tool that can greatly improve the efficiency of the upgrade. I'm currently coordinating with him to verify the details, and once everything is confirmed, we can accelerate the migration from JUnit 4 to JUnit 5.

Reply:
@pjfanning Can you share any compatibility issues you've encountered? I personally haven't noticed any. Of course, there are a few unit test failures under JDK 11, but I will be submitting a fix in a PR soon.

Reply:
If we've prepared the trunk code for compatibility with JDK 17, including unit tests and CI, do we still need to release hadoop-3.4.2, or can we directly release hadoop-3.5.0 based on the trunk?

Reply:
I think a 3.4.2 can go out anyway, so people on java <17 can upgrade with lower risk dependency updates etc, and java8+

Reply:
I support your point of view. We indeed need to keep two versions (hadoop-3.4.2 and hadoop-3.5.0) until JDK 17 becomes stable enough.

Reply:
> I would like to ask whether we need to backport this([Hadoop-15984](https://issues.apache.org/jira/browse/HADOOP-15984)) to branch-3.4, or should we release it in [hadoop-3](https://issues.apache.org/jira/browse/HADOOP-3).5.0?

Thanks for your works. IMO, it's not necessary to backport Hadoop-15984 to other active line. In my opinion support to JDK17 should be as one new feature with release-3.5 line, and other release line keep the current status. We will launch discuss about the first release version of line 3.5 when HDFS-17384([HDFS NameNode FGL) and HDFS-17531(Asynchronous router RPC) are ready. What do you think about?

Reply:
@Hexiaoqiao Thank you for your message. I agree with your point. We will keep HADOOP-15984 in the trunk branch and avoid updating it in the active branches to continue verifying this change. Once the other features are ready, we can discuss the release plan for Hadoop 3.5.0 together. In the meantime, I will continue to follow up on the JDK 17 upgrade and the necessary improvements, hoping we can successfully achieve our goals.